resolver/namerecord.go

```diff
@@ -88,6 +88,10 @@ func GetNameRecord(domain, question string) (*NameRecord, error) {
 
 // DeleteNameRecord deletes a NameRecord from the database.
 func DeleteNameRecord(domain, question string) error {
+	// In order to properly delete an entry, we must also clear the caches.
+	recordDatabase.FlushCache()
+	recordDatabase.ClearCache()
+
 	key := makeNameRecordKey(domain, question)
 	return recordDatabase.Delete(key)
 }
```

Comment:
This get's called everytime a rrCache record is expired and we are going to clear the whole cache everytime this happens. We should check how much performance we are going to lose here if we clear the cache and start refilling it over and over again.

Reply:
This function is only used for resetting the cache for a single domain/query - I improved the name to better reflect that.